pyrk/th_component.py

```diff
@@ -83,33 +131,232 @@ def update_temp(self, timestep, temp):
+
+    def addConvBC(self, env, prev_comp, h, R):
+        '''add convective boundary condition
+        :param env: name of the environment for convective heat transfer
+        (the fluid)
+        :type env: str
+        :param prev_comp: name of the component that is immediately inside the
+        boundary component
+        :type prev_comp: str
+        :param h: convective heat transfer coefficient
+        :type h: float
+        :param R: radius of the sphere
+        :type R: float
+        '''
+        self.convBC[env] = {
+            "h": h.to('joule/second/kelvin/meter**2'),
+            "prev_comp": prev_comp,
+            "R": R
+        }
+
+    def add_conduction(self, env, k,
+                       area=0.0*units.meter**2, L=0.0*units.meter,
+                       r_b=0.0*units.meter, r_env=0.0*units.meter):
+        '''Add parameters for conduction heat transfer calculation
+        area and L are used for slab geometry
+        r_b and r_env are used for spherical heat diffusion
+
+        :param env: name of the component that this component conduct heat to
+        :type env: str
+        :param k: thermal conductivity of the material that heat is conducted in(may not be the 'self' component)
+        :type k:float
+        :param area: conduction surface for the slab
+        :type area: float
+        :param L: thickness of the slab
+        :type L: float
+        :param r_b: outer radius of the component
+        :type r_b: float
+        :param r_env: outer radius of the environment
+        :type r_env: float
+        '''
+        self.cond[env] = {
+            "k": k.to('watts/meter/kelvin'),
+            "area": area.to('meter**2'),
+            "L": L.to('meter'),
+            "r_b": r_b.to('meter'),
+            "r_env": r_env.to('meter')
+        }
+
+    def add_advection(self, name, m_flow, t_in, cp):
+        '''Add advection dictionary to the fluid component(coolant) that has
+        advective heat tranfer
+
+        :param m_flow: mass flow rate
+        :type m_flow: float
+        :param t_in: temperature at the inlet of the control volume
+        :type t_in: float
+        :param cp: specific heat capacity
+        :type cp: float
+        '''
+        self.adv[name] = {
+            "m_flow": m_flow.to('kg/second'),
+            "t_in": t_in.to('kelvin'),
+            "cp": cp.to('joule/kg/kelvin')
+        }
+
+
+class THSuperComponent(object):
+
+    '''A 'component' containing a list of component
+    Creating a superComponent would automatically define conduction between the
+    mesh elements'''
+
+    def __init__(self, name, T0, sub_comp=[], timer=Timer()):
+
+        """Initalizes a thermal hydraulic super component.
+
+        :param name: The name of the supercomponent (i.e., "fuel" or "cool")
+        :type name: str.
+        :param T0: The initial temperature of the supercomponent
+        :type T0: float.
+        :param sub_comp: List of components that makes up the supercomponent.
```

Comment:
Just a suggestion: sub_comp might be better if it were called sub_comps so that we know it's a list when we read the variable name (because it's plural)